server/infinispan/src/main/java/org/hawkular/apm/server/infinispan/InfinispanSpanCache.java

```diff
@@ -114,12 +104,18 @@ public void store(String tenantId, List<Span> spans, Function<Span, String> cach
         }
 
         for (Span span : spans) {
-
-            if (log.isLoggable(Level.FINEST)) {
-                log.finest("Store span [id=" + span.getId() + "]: " + span);
-            }
+            log.debugf("Store span [%s]" + span);
 
             spansCache.put(cacheKeyEntrySupplier.apply(span), span, 1, TimeUnit.MINUTES);
+
+            if (span.getTraceId() != null) {
+                Set<Span> trace = traceCache.get(span.getTraceId());
+                if (trace == null) {
+                    trace = new HashSet<>();
+                    traceCache.put(span.getTraceId(), trace);
```

Comment:
Need to 'put' in the cache, after the trace set has been updated - the put may result in the cached object being written to disk or distributed to remote node if we change the cache type.